compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRArgument.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.ballerinalang.compiler.bir.model;
+
+/**
+ * {@code BIRArgument} represents an argument in an instruction.
+ *
+ * @since 2.0.0
+ */
+public class BIRArgument extends BIROperand {
+
+    public ArgumentState argState;
+    public BIROperand userProvidedCondition;
+
+    public BIRArgument(ArgumentState argState, BIROperand argVar) {
```

Comment:
If Argument extends BIROperand, why do we need still get the BIROperand in the constructor ?

Reply:
I did this because we get argument as `BIROperand` type. (Within the constructor I used it as `BIRVariableDcl`). It can be changed to `BIRVariableDcl` type. I will fix it.

Reply:
If you check the usage of the constructor, we are actually wrapping the BIROperand around BIRArgument, which is not needed IMO, if you can directly pass in the varDcl.

Reply:
Fixed.